.github/workflows/merge-queue-metric.yml

```diff
@@ -0,0 +1,42 @@
+# This workflow tracks when PRs are added ('enqueued') to the master branch merge queue, and when they are removed for
+# reasons other than 'MERGE', which means that it probably has to be enqueued again. The most common reason for this is
+# test failures when running the checks in the merge queue.
+name: Tracking Merge Queue
+on:
+  pull_request:
+    types: [ "enqueued", "dequeued" ]
+    branches:
+      - "master"
+
+permissions:
+  id-token: write
+
+jobs:
+  emit-enqueue:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'aws/aws-sdk-java-v2' && github.event.action == 'enqueued' }}
+
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.CI_AWS_ROLE_ARN }}
+          aws-region: us-west-2
+          role-duration-seconds: 120
+      - name: Record merge queue add
+        run: |
+          aws --region us-west-2 cloudwatch put-metric-data --namespace AwsJavaSdkV2/GitHub --metric-name MergeQueue-Add --unit Count --value 1 --dimensions Branch=master
```

Comment:
Suggestion: I suggest renaming the CloudWatch metrics to provide better context about the events being tracked when looked in aws console outside of github. Some suggestions

- GitHub_MergeQueue_Enqueued
-  GitHub_MergeQueue_Add
- GitHub_MergeQueue_Dequeued
- GitHub_MergeQueue_Removed
- GitHub_MergeQueue_Abandoned

Reply:
Hmm it should already be clear from the namespace that's related to GitHub right? `AwsJavaSdkV2/GitHub`.

I don't want to track too many events for simplicity; if we broke it down to finer metrics, we'd have to do metric math to see things in aggregate, which is really what we want atm. We can decide later if we want to track finer grained metrics.

I didn't name it `MergeQueue-Dequeue`/`Enqueued` because if the type is `dequeue`, it's not necessarily a bad thing. If the PR is merged into master, that is also a `dequeue`.